docs/java-rest/high-level/apis/update.asciidoc

```diff
@@ -0,0 +1,196 @@
+--------------------------------------------------
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[update-request-scripted-upsert]
+--------------------------------------------------
+<1> Indicate that the script must run regardless of whether the document exists or not,
+ie the script takes care of creating the document if it does not already exist.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[update-request-doc-upsert]
+--------------------------------------------------
+<1> Indicate that the partial document must be used as the upsert document if it
+does not exist yet.
+
+[[java-rest-high-document-update-sync]]
+==== Synchronous Execution
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[update-execute]
+--------------------------------------------------
+
+[[java-rest-high-document-update-async]]
+==== Asynchronous Execution
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[update-execute-async]
+--------------------------------------------------
+<1> Called when the execution is successfully completed. The response is
+provided as an argument.
+<2> Called in case of failure. The raised exception is provided as an argument.
+
+[[java-rest-high-document-update-response]]
+==== Update Response
+
+The returned `UpdateResponse` allows to retrieve information about the executed
+ operation as follows:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[update-response]
+--------------------------------------------------
+<1> Handle the case where the document was created for the first time
+<2> Handle the case where the document was updated
+<3> Handle the case where the document was deleted
+<4> Handle the case where the document was not impacted by the update,
+ie no operation (noop) was executed on the document
+
+
+It is also possible to check for shard failures:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[update-failure]
+--------------------------------------------------
+<1> Handle the situation where number of successful shards is less than
+total shards
+<2> Handle the potential failures
+
+
+If there is a version conflict, an `ElasticsearchException` will
+be thrown:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[update-conflict]
+--------------------------------------------------
+<1> The raised exception indicates that a version conflict error was returned.
```

Comment:
can we add docs for `fetchSource` (you can borrow it from my get api PR), `retryOnConflict` and `waitForActiveShards` ?